File: app/src/main/java/com/example/amconsofttest/ui/BaseActivity.java
package com.example.amconsofttest.ui;

import android.annotation.SuppressLint;
import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.widget.Toast;

import com.arellomobile.mvp.MvpAppCompatActivity;

@SuppressLint ("Registered")
public class BaseActivity extends MvpAppCompatActivity implements BaseView {
    @Override
    public void showToast(int id) {
        Toast.makeText(this, getText(id), Toast.LENGTH_LONG).show();
    }

    public void replaceFragment(@IdRes int containerViewId,
                                @NonNull Fragment fragment,
                                String fragmentTag) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(containerViewId, fragment, fragmentTag)
                .addToBackStack("")
                .commitAllowingStateLoss();
    }

    @Override
    public void onBackPressed() {
        if (getSupportFragmentManager().getBackStackEntryCount() > 1) {
            getSupportFragmentManager().popBackStack();
        } else {
            finish();
        }
    }
}

File: app/src/main/java/com/example/amconsofttest/ui/BaseActivity.kt
package com.example.amconsofttest.ui

import android.annotation.SuppressLint
import android.support.annotation.IdRes
import android.support.v4.app.Fragment
import android.widget.Toast

import com.arellomobile.mvp.MvpAppCompatActivity

@SuppressLint("Registered")
open class BaseActivity : MvpAppCompatActivity(), BaseView {
    override fun showToast(id: Int) {
        Toast.makeText(this, getText(id), Toast.LENGTH_LONG).show()
    }

    fun replaceFragment(@IdRes containerViewId: Int,
                        fragment: Fragment,
                        fragmentTag: String) {
        supportFragmentManager
                .beginTransaction()
                .replace(containerViewId, fragment, fragmentTag)
                .addToBackStack("")
                .commitAllowingStateLoss()
    }

    override fun onBackPressed() {
        if (supportFragmentManager.backStackEntryCount > 1) {
            supportFragmentManager.popBackStack()
        } else {
            finish()
        }
    }
}
